packages/charts/src/charts/line/step.component.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { motion } from 'framer-motion';
+
+import { StepType, Mark } from './types';
+
+import { findMarksInCluster } from './line-chart.utils';
+
+const pointMotion = {
+  initial: 'hidden',
+  animate: 'show',
+  variants: { hidden: { opacity: 0 }, show: { opacity: 1 } },
+  transition: { delay: 0.1, duration: 0.3 },
+};
+
+type Props = {
+  steps: StepType[];
+  marks: Record<number, Mark[]>;
+  onMouseEnter: (e: React.MouseEvent, mark: StepType) => void;
+  onMouseMove: (e: React.MouseEvent, mark: StepType) => void;
+  onMouseLeave: (e: React.MouseEvent) => void;
+};
+
+const Step = ({
+  steps,
+  marks,
+  onMouseEnter,
+  onMouseMove,
+  onMouseLeave,
+}: Props) => {
+  const [activeStep, setActiveStep] = useState(null);
+  const filteredMarks =
+    activeStep && findMarksInCluster(activeStep, marks, activeStep.height);
+  return (
+    <>
+      {filteredMarks &&
+        filteredMarks.map((mark: Mark) => {
+          const { key, color, radius, x, y } = mark;
+          return (
+            <g key={key}>
+              <motion.circle
+                cx={x}
+                cy={y}
+                r={radius}
+                fill={color}
+                {...pointMotion}
+              />
+            </g>
+          );
+        })}
+      {steps !== undefined &&
+        steps.map((item: StepType) => {
+          const itemTransfomed = {
```

Comment:
typo and please consider reverting adjective with noun `itemTransfomed ` => `transformedItem ` to keep common naming conventions.